packages/knip/src/reporters/gitlab.ts

```diff
@@ -0,0 +1,128 @@
+import { createHash } from 'node:crypto';
+import type { Entries } from 'type-fest';
+import type { Issue, IssueRecords, IssueSeverity, IssueSymbol, Report, ReporterOptions } from '../types/issues.js';
+import { getTitle } from './util.js';
+
+/**
+ * Entry for GitLab Code Quality report. See: https://docs.gitlab.com/ci/testing/code_quality/#code-quality-report-format
+ */
+interface CodeQualityEntry {
+  check_name: string;
+  description: string;
+  severity: 'info' | 'minor' | 'major' | 'critical' | 'blocker';
+  fingerprint: string;
+  location: {
+    path: string;
+    lines: {
+      begin: number;
+    };
+  };
+}
+
+export default async ({ report, issues }: ReporterOptions) => {
+  const entries: CodeQualityEntry[] = [];
+  const hashes = new Set<string>();
+
+  for (const [type, isReportType] of Object.entries(report) as Entries<Report>) {
+    if (!isReportType) {
+      continue;
+    }
+
+    if (type === 'files' || type === '_files') {
+      continue;
+    }
+
+    for (const issue of flatten(issues[type] as IssueRecords)) {
+      const { filePath } = issue;
+
+      if (type === 'duplicates' && issue.symbols) {
+        entries.push(
+          ...issue.symbols.map(symbol => ({
+            check_name: getTitle(type),
+            description: getSymbolDescription({ symbol, parentSymbol: issue.parentSymbol }),
+            severity: convertSeverity(issue.severity),
+            fingerprint: createFingerprint(filePath, symbol.symbol, hashes),
+            location: {
+              path: filePath,
```

Comment:
According to the specs, the `filePath` should be relative, you can import the [toRelative](https://github.com/webpro-nl/knip/blob/7e22547039df179a47335c4be25736acf0e5ad33/packages/knip/src/util/path.ts#L30) helper for this.